### PR TITLE
Add some more media sites

### PR DESCRIPTION
**Pull request by iMattPro**

### CODE REVIEW

File: collection/sites/mastodon.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: "Mastodon"
+host: mastodon.social
+example: https://mastodon.social/@HackerNewsBot/100181134752056592
+extract: "!//(?'host'[-.\\w]+)/@(?'name'\\w+)/(?'id'\\d+)!"
+oembed:
+    endpoint: https://mastodon.social/api/oembed
+    scheme: https://mastodon.social/@{@name}/{@id}
+scrape:
+    - extract:
+        url: "!https://(?'host'[-.\\w]+)/@(?'name'\\w+)/(?'id'\\d+)!"
```
Comment (rxu) [inline collection/sites/mastodon.yml:10]: @JoshyPHP could you please judge if this part of definition is fully correct, especially the inported part of `extract` which looks like `'extract'=>['#"url":"https://(?\'host\'[-.\\w]+)/@(?\'name\'\\w+)/(?\'id\'\\d+)"#']` in `CachedDefinitionCollection.php`.

Comment (JoshyPHP) [reply collection/sites/mastodon.yml:10]: At first glance it looked fine but on a second read I realized it was a trick of the eye caused by the YML structure. The regexp I use is `#"url":"https://(?'host'[-.\w]+)/@(?'name'\w+)/(?'id'\d+)"#`. That code looks like it creates a `url` attribute inside the `extract` definition.

I should mention that `CachedDefinitionCollection.php` is auto-generated from [XML files](https://github.com/s9e/TextFormatter/tree/master/src/Plugins/MediaEmbed/Configurator/sites). The XML files are canonical and easier to read.

Comment (iMattPro) [reply collection/sites/mastodon.yml]: So... Like the way I just updated it is how it should be?

Comment (JoshyPHP) [reply collection/sites/mastodon.yml]: That looks like it. If there's anything wrong with it, I can't see it.

**Comment by rxu** [comment]
Bilibili doesn't load in the post for some reason giving just an empty space (but using the link directly in the browser works fine).
image

Tenor loads with the header (not just a GIF), but that's probably like it should be here.
image

**Comment by iMattPro** [comment]
For Tenor it loads the GIF fine for me, the header only appears when you mouse over the GIF for me.

For Bilibili, being a Chinese site, maybe there's some country issue there for you??? The example working for me.
Screenshot 2023-03-07 at 8 02 14 PM

**Comment by rxu** [comment]
Well, regional issue would be the case for Bilibili but example link loads fine directly in the browser. Also the site works on my side in common. Weird.

**Comment by iMattPro** [comment]
Yeah I don't know. I've tried in all 4 browsers and they work here (Edge, Firefox, Chrome and Safari)

**Review by rxu — APPROVED** [review]

**Pull request merged**